src/core/TrainController.hpp

```diff
@@ -99,26 +97,23 @@ namespace godot {
             void set_train_id(const String &train_id);
             String get_type_name() const;
             void set_type_name(const String &type_name);
-            void set_battery_voltage(const double p_value);
+            void set_battery_voltage(double p_value);
             double get_battery_voltage() const;
             void set_mass(double p_mass);
             double get_mass() const;
             void set_power(double p_power);
             double get_power() const;
             void set_max_velocity(double p_value);
             double get_max_velocity() const;
-            void set_axle_arrangement(String p_value);
+            void set_axle_arrangement(const String &p_value);
             int get_radio_channel_min() const;
-            void set_radio_channel_min(const int p_value);
+            void set_radio_channel_min(int p_value);
             int get_radio_channel_max() const;
-            void set_radio_channel_max(const int p_value);
+            void set_radio_channel_max(int p_value);
             String get_axle_arrangement() const;
 
-            void set_state(const Dictionary &p_state);
+            void set_state(const Dictionary &p_state) {};
```

Comment:
This should not be implemented as empty method. This is unused method, so declaration can be safely removed.